Tests for Trezor and Coldcard implementations and enable Travis CI

achow101 opened the pull request:
This PR is based on top of #66. It adds two test cases for the Trezor and Coldcard implementations by using their respective emulators/simulators. `test_bip32.py` was removed as the BIP 32 implementation was removed and no longer needed. Additionally a test running (`run_tests.py`), a script for building the Trezor emulator, Coldcard simulator, and bitcoind, and instructions for running the tests were all added. Lastly a Travis config has been added and travis builds for this branch are available at https://travis-ci.org/achow101/HWI

instagibbs commented on test/test_coldcard.py:
```diff
@@ -0,0 +1,252 @@
+
+# Setup bitcoind with no privkey wallet and some blocks
+rpc.generatetoaddress(101, rpc.getnewaddress())
+rpc.createwallet('trezor_test', True)
```
s/trezor_test/coldcard_test/ everywhere in this file

achow101 replied:
Fixed

instagibbs commented on test/test_coldcard.py:
```diff
@@ -0,0 +1,252 @@
+assert(import_result[0]['success'])
+for i in range(0, 21):
+    addr_info = wrpc.getaddressinfo(wrpc.getnewaddress())
+    assert(addr_info['hdkeypath'] == 'm/49\'/1\'/3\'/0/{}'.format(i))
```
just use `"` around it so you don't have to escape the hardening chars in this test file

achow101 replied:
Done

achow101 commented:
I've updated `setup_environment.sh` so that it can make better use of the caching the travis is setup to do.

achow101 commented on test/README.md:
````diff
@@ -0,0 +1,56 @@
+```
+pipenv
+```
+
````
You can build it without docker, unless you want to use `./build.sh`. AFAIK, `build.sh` does not build the emulator in headless mode, so there's a GUI and such. For automated testing on Travis, we need headless mode so we build it through the `pipenv` commands rather than `build.sh`.

achow101 commented on test/README.md:
````diff
@@ -0,0 +1,56 @@
+The following python packages will need to be installed
+
+```
+pipenv
````
Done

achow101 commented on test/README.md:
```diff
@@ -0,0 +1,56 @@
+
+## Coldcard simulator dependencies
+
+In order to build the Coldcard simulator, the following packages will need to be installed:
```
Done

achow101 commented:
I can add another test that uses the Trezor Core emulator.

I've updated the README significantly with more detailed instructions for building. I've also updated the shebang for `test_bech32.py`.

`setup.py` will install `python-bitcoinrpc` if you install the `tests` extra.

Sjors commented:
@achow101 might as well add those Trezor Core tests in a next PR. I can always look at Travis logs to see if I broke a test.
